### PR TITLE
NAS-106059 / 12.0 / NAS-106059: Fix pool.processes raising validation error

### PR DESCRIPTION
**Pull request by bastiao**

### CODE REVIEW

File: src/middlewared/middlewared/plugins/pool.py [hunk 1]
```diff
@@ -3577,7 +3577,21 @@ async def processes(self, oid):
         ]
         """
         result = []
-        dataset = await self.get_instance(oid)
+        dataset = None
+        path=None
```
Comment (themylogin) [inline src/middlewared/middlewared/plugins/pool.py:3581]: PEP8 checked did not catch this for some reason

**Comment by sonicaj** [comment]
I think we should check if root dataset can be found in `pool.processes` and not `pool.dataset.processes` and if it can't be found, we return an empty list from there ( `pool.processes` ). `pool.dataset.processes` should raise an error if the dataset in question does not exist. Thoughts @themylogin ?

**Comment by themylogin** [comment]
I agree with you @sonicaj

**Comment by william-gr** [comment]
@bastiao please go ahead and merge using Squash button like Vladimir said.

Thank you

**Pull request merged**